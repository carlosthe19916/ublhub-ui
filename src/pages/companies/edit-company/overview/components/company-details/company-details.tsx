import React from "react";
import {
  Card,
  CardTitle,
  CardBody,
  DescriptionList,
  DescriptionListGroup,
  DescriptionListTerm,
  DescriptionListDescription,
} from "@patternfly/react-core";
import { Company } from "api/models";

export interface CompanyDetailsProps {
  company: Company;
}

export const CompanyDetails: React.FC<CompanyDetailsProps> = ({ company }) => {
  return (
    <Card>
      <CardTitle>Detalle</CardTitle>
      <CardBody>
        <DescriptionList>
          <DescriptionListGroup>
            <DescriptionListTerm>RUC</DescriptionListTerm>
            <DescriptionListDescription>
              {company.ruc}
            </DescriptionListDescription>
          </DescriptionListGroup>
          <DescriptionListGroup>
            <DescriptionListTerm>Nombre</DescriptionListTerm>
            <DescriptionListDescription>
              {company.name}
            </DescriptionListDescription>
          </DescriptionListGroup>
          <DescriptionListGroup>
            <DescriptionListTerm>Descripci√≥n</DescriptionListTerm>
            <DescriptionListDescription>
              {company.description}
            </DescriptionListDescription>
          </DescriptionListGroup>
        </DescriptionList>
      </CardBody>
    </Card>
  );
};
