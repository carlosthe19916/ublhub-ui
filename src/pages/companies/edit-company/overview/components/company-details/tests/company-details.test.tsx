import React from "react";
import { shallow } from "enzyme";
import { CompanyDetails } from "../company-details";

describe("CompanyDetails", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(
      <CompanyDetails
        company={{
          name: "my company",
          description: "my description",
          webServices: {
            factura: "http://url1.com",
            guia: "http://url2.com",
            retenciones: "http://url3.com",
          },
          credentials: {
            username: "myUsername",
            password: "myPassword",
          },
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
