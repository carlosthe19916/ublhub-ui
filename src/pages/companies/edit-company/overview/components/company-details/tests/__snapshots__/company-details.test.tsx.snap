// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyDetails Renders without crashing 1`] = `
<Card>
  <CardTitle>
    Detalle
  </CardTitle>
  <CardBody>
    <DescriptionList>
      <DescriptionListGroup>
        <DescriptionListTerm>
          RUC
        </DescriptionListTerm>
        <DescriptionListDescription />
      </DescriptionListGroup>
      <DescriptionListGroup>
        <DescriptionListTerm>
          Nombre
        </DescriptionListTerm>
        <DescriptionListDescription>
          my company
        </DescriptionListDescription>
      </DescriptionListGroup>
      <DescriptionListGroup>
        <DescriptionListTerm>
          Descripci√≥n
        </DescriptionListTerm>
        <DescriptionListDescription>
          my description
        </DescriptionListDescription>
      </DescriptionListGroup>
    </DescriptionList>
  </CardBody>
</Card>
`;
